### PR TITLE
Fix issue 600

### LINKED ISSUE #600: RAW: TypeError when omitting detector_size
To reproduce (file doesn't need to exist):

```python
ctx.load("raw", path="/tmp/meh/meh.raw", dtype="fliat32", scan_size=(256, 256))
```

result:

```
TypeError: 'NoneType' object is not iterable
```

Should instead throw an understandable error message.

### PR DESCRIPTION
**Pull request by twentyse7en**
Better error message when parameter detector_size is missing.
Fix #600 
## Contributor Checklist:

* [ ] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated test cases

### CODE REVIEW

**Comment by twentyse7en** [comment]
is bugfix folder the right place to add changelog entry for this issue ?

File: src/libertem/io/dataset/raw.py [hunk 1]
```diff
@@ -201,6 +202,9 @@ def __init__(self, path, scan_size, dtype, detector_size=None, enable_direct=Fal
                                  "please use EMPAD DataSet")
             detector_size = crop_detector_to
 
+        if detector_size is None:
+            raise ValueError("missing 1 required argument: 'detector_size'")
```
Comment (sk1p) [inline src/libertem/io/dataset/raw.py]: This should be a `TypeError`; see the descriptions on https://docs.python.org/3/library/exceptions.html, especially:

```
exception ValueError
Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError
```

Also, as you noted in #600, `TypeError` is also the exception type you get in all other cases if you omit a required parameter.

**Comment by sk1p** [comment]
> is bugfix folder the right place to add changelog entry for this issue ?

Yes!

File: tests/test_context.py [hunk 2]
```diff
@@ -6,3 +10,28 @@ def test_ctx_load(lt_ctx, default_raw):
...
+
+def test_missing_detector_size(lt_ctx, default_raw):
+    with pytest.raises(ValueError) as e:
+        lt_ctx.load(
+            "raw",
+            path=default_raw._path,
+            scan_size=(16, 16),
+            dtype="float32",
+            )
+    assert e.match("missing 1 required argument: 'detector_size'")
```
Comment (sk1p) [inline tests/test_context.py]: Tests look good! They should be moved to `tests/io/test_raw.py` though. You may only be calling the `load` method of the `Context`, but the behavior you are testing comes from the raw dataset.

**Comment by twentyse7en** [comment]
Added changes as requested.

File: tests/test_context.py (hunk 2 above)
Comment (twentyse7en) [reply tests/test_context.py]: I thought of adding the tests to `test_raw.py` at first. But when I search for the test which open raw file normal way, It was in `test_context.py`. So I got confused.

Comment (sk1p) [reply tests/test_context.py]: Ah, I can see how that is confusing. All the tests in `test_raw` just use fixtures like `default_raw` and there was no test yet that exercised the different parameters. Looks good now!

**Comment by twentyse7en** [comment]
Added changelog entry as you suggested.
 
> If you like, you can also update your description in `creators.json` - you have by now done a bit more than just fixing issue 81!

Thx :) I will update in some other pull request.
Sorry for late reply, I'm having internal exam this week. I couldn't invest enough time as I thought of.

**Comment by sk1p** [comment]
> Added changelog entry as you suggested.

:+1: 

> Thx :) I will update in some other pull request.

Ok :+1:

> Sorry for late reply, I'm having internal exam this week. I couldn't invest enough time as I thought of.

All okay! Thanks, will merge now!

**Pull request merged**